ReceiptRepository/Tests/PrepaidCardReceiptRepositoryTests.swift

```diff
@@ -0,0 +1,90 @@
+
+import Hippolyte
+import HyperwalletSDK
+@testable import ReceiptRepository
+import XCTest
+
+class PrepaidCardReceiptRepositoryTests: XCTestCase {
+    private lazy var listPrepaidCardReceiptPayload = HyperwalletTestHelper.getDataFromJson("PrepaidCardReceiptResponse")
+    private var factory: ReceiptRepositoryFactory!
+    private var prepaidCardReceiptRepository: PrepaidCardReceiptRepository!
+    private var receiptExpectation: XCTestExpectation!
+
+    override func setUp() {
+        Hyperwallet.setup(HyperwalletTestHelper.authenticationProvider)
+        factory = ReceiptRepositoryFactory.shared
+        prepaidCardReceiptRepository = factory.prepaidCardReceiptRepository()
+        receiptExpectation = self.expectation(description: "load prepaid card receipts")
+    }
+
+    override func tearDown() {
+        ReceiptRepositoryFactory.clearInstance()
+        if Hippolyte.shared.isStarted {
+            Hippolyte.shared.stop()
+        }
+    }
+
+    func testListPrepaidCardReceipt_success() {
+        var prepaidCardReceiptList = [HyperwalletReceipt]()
+        HyperwalletTestHelper.setUpMockServer(
+            request: ReceiptRequestHelper.setUpRequest(listPrepaidCardReceiptPayload,
+                                                       nil,
+                                                       "trm-123456789"))
+        prepaidCardReceiptRepository.listPrepaidCardReceipts(
+            prepaidCardToken: "trm-123456789"
+        ) { [weak receiptExpectation ] result in
+            switch result {
+            case .success(let receiptPageList):
+                guard let receiptPageList = receiptPageList else {
+                    XCTFail("The Prepaid Card receipt list should not be empty")
+                    return
+                }
+                prepaidCardReceiptList = receiptPageList.data
+                receiptExpectation?.fulfill()
+
+            case .failure:
+                XCTFail("Unexpected error")
+            }
+        }
+        wait(for: [receiptExpectation], timeout: 1)
+        XCTAssertFalse(prepaidCardReceiptList.isEmpty, "The Prepaid Card receipt list should not be empty")
+    }
+
+    func testListPrepaidCardReceipt_failure() {
+        HyperwalletTestHelper.setUpMockServer(
+            request: ReceiptRequestHelper.setUpRequest(listPrepaidCardReceiptPayload,
+                                                       NSError(domain: NSURLErrorDomain, code: 500, userInfo: nil),
+                                                       "trm-123456789"))
+        prepaidCardReceiptRepository.listPrepaidCardReceipts(
+            prepaidCardToken: "trm-123456789"
+        ) { [weak receiptExpectation ] result in
+            switch result {
+            case .success:
+                XCTFail("The listPrepaidCardReceipts method should return Error")
+
+            case .failure(let error):
+                let originalError = ErrorHelper.parseOriginalError(error)
```

Comment:
This is one way of doing it or you can simply XCTAssertNotNil(error, "Error should not be nil"). No need of parsing it to get the original error. You can remove that ErrorHelper class as well

Reply:
good point, fixed